### PR TITLE
Couple build fixes

### PR DESCRIPTION
**Pull request by marler8997**

### CODE REVIEW

**Review by wilzbach — APPROVED** [review]

File: src/build.d [hunk 1]
```diff
@@ -209,7 +209,7 @@ alias lexer = memoize!(function()
             "-of" ~ target,
             "-lib",
             "-vtls",
-            "-J"~env["G"], "-J../res",
+            "-J" ~ env["RES"],
```
Comment (PetarKirov) [inline src/build.d]: As far as I can see there are only two files that are string-imported (and for which we need the `-J` flag is needed):
* `VERSION`, used by https://github.com/dlang/dmd/blob/a7017a9634a33d6234d28b60bc96a1a09e5d704c/src/dmd/globals.d#L355 and produced by the `versionFile` target and placed in `env["G"]`. `globals.d` is used by the `lexer` target, so I think we need to keep it.
* `res/default_ddoc_theme.ddoc`, used by https://github.com/dlang/dmd/blob/a7017a9634a33d6234d28b60bc96a1a09e5d704c/src/dmd/doc.d#L366
`doc.d` is part of the `frontend` source, but not of the `lexer` ones, so I think we should drop `env["RES"]` here.

File: src/build.d [hunk 2]
```diff
@@ -357,8 +357,7 @@ alias dmdExe = memoize!(function(string targetSuffix, string[] extraFlags...)
             env["HOST_DMD_RUN"],
             "-of" ~ target,
             "-vtls",
-            "-J"~env["G"],
```
Comment (PetarKirov) [inline src/build.d]: We need `"-J" ~ env["G"]` to be able to string-import the `VERSION` file, or am I missing something?

**Review by PetarKirov — CHANGES_REQUESTED** [review]

File: src/build.d (hunk 1 above)
Comment (wilzbach) [reply src/build.d]: `globals.d` is not imported by the lexer. That's why the import can be removed here.

Comment (PetarKirov) [reply src/build.d]: But why is it part of the `lexer` sources then?

File: src/build.d (hunk 2 above)
Comment (MoonlightSentinel) [reply src/build.d]: Its included in `flags["DFLAGS"`, see https://github.com/dlang/dmd/blob/cc78c274384608098a2a8ca25984903cb69078ef/src/build.d#L879

File: src/build.d (hunk 1 above)
Comment (marler8997) [reply src/build.d]: Ok I've removed the res import path from `lexer` and it still appears to build.  Nice catch.

**Review by PetarKirov — APPROVED** [review]

**Pull request merged**